src/squeue.rs

```diff
@@ -20,14 +20,11 @@ pub struct SubmissionQueue {
     pub(crate) sqes: *mut sys::io_uring_sqe,
 }
 
-/// A snapshot of the submission queue.
-pub struct AvailableQueue<'a> {
-    head: u32,
-    tail: u32,
-    ring_mask: u32,
-    ring_entries: u32,
-    queue: &'a mut SubmissionQueue,
-}
+/// A type that grants mutable access to an io_uring's [submission queue](SubmissionQueue).
+///
+/// This is necessary to prevent users swapping out the submission queue, which can cause
+/// unsoundness.
+pub struct Mut<'a>(pub(crate) &'a mut SubmissionQueue);
```

Comment:
I think we can keep the name of `SubmissionQueue` and make the original `SubmissionQueue` a private structure. to avoid excessive API changes.